React/wbooks/src/app/components/RouteHandler/index.js

```diff
@@ -1,26 +1,30 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
 import { Route } from 'react-router-dom';
 import { Redirect } from 'react-router';
 
 import Navbar from '../Navbar/index.js'
 import * as localstorageService from '../../../services/localstorageService';
+import * as booksActions from '../../../redux/books/actions';
+import * as sessionActions from '../../../redux/session/actions';
 
 class RouteHandler extends Component {
-  validateAuth = props => {
-    return localstorageService.retrieveUserTokenFromLocalStorage();
-  };
-
-  isPublicRequest = props => {
-    return props.location.pathname === '/' || props.location.pathname === '/signup';
-  };
+  componentWillMount = () => {
+    const user = localstorageService.retrieveUserTokenFromLocalStorage();
+    if(user) {
+      this.props.loadBooks();
+      this.props.loadSession();
+    }
+  }
```

Comment:
U re introducing tasks that doesn't correspond to the scope of component. That's not right. U must dispatch just an action, an these action do all these tasks. This actions must be dispatched at the highest level. Where would you put it?